base/setup/lib/utils/partlist.c

```diff
@@ -192,21 +179,16 @@ AssignDriveLetters(
         {
             PartEntry = CONTAINING_RECORD(Entry2, PARTENTRY, ListEntry);
 
-            PartEntry->DriveLetter = 0;
+            if (!PartEntry->Volume)
+                continue;
+            PartEntry->Volume->Info.DriveLetter = 0;
 
-            if (PartEntry->IsPartitioned)
+            if (PartEntry->IsPartitioned &&
+                (IsRecognizedPartition(PartEntry->PartitionType) ||
+                 PartEntry->SectorCount.QuadPart != 0LL))
             {
-                ASSERT(PartEntry->PartitionType != PARTITION_ENTRY_UNUSED);
-
-                if (IsRecognizedPartition(PartEntry->PartitionType) ||
-                    PartEntry->SectorCount.QuadPart != 0LL)
-                {
-                    if (Letter <= L'Z')
-                    {
-                        PartEntry->DriveLetter = Letter;
-                        Letter++;
-                    }
-                }
+                if (Letter <= L'Z')
+                    PartEntry->Volume->Info.DriveLetter = Letter++;
```

Comment:
Same question here. Does this match Windows?

Reply:
https://github.com/reactos/reactos/pull/7159#discussion_r1724825665